llvm/lib/ExecutionEngine/Interpreter/Interpreter.h

```diff
@@ -93,6 +93,47 @@ class Interpreter : public ExecutionEngine, public InstVisitor<Interpreter> {
 
   bool CleanAbort;
 
+protected:
+  // Those function will be used through PartialInterpreter interface
+  // to manipulate the call frame stack
+  bool isAlreadyInCallStack(const llvm::Function* F) const
+  {
+    for (auto& x : ECStack)
+      if (x.CurFunction == F)
+        return true;
+    return false;
+  }
+  bool isBottomStack() const
```

Comment:
isInitialStack maybe? this is the execution stack of the function we are currently partially execution (while entry point to me means the actual callsite)
Actually stack is probably the wrong terminology, since ECStack is a stack of execution contexts, so probably all these functions could be named isFrontExecutionContext or isFrontExecContext